test/SlowTests/Issues/RavenDB_15409.cs

```diff
@@ -12,15 +18,67 @@ namespace SlowTests.Issues
 {
     public class RavenDB_15409 : ClusterTestBase
     {
+        private class DummyWebSocket : WebSocket
+        {
+            private static WebSocketReceiveResult Result { get; } = new(1, WebSocketMessageType.Text, true);
+            private readonly TaskCompletionSource<WebSocketReceiveResult> _completionSource = new();
+            private readonly MemoryStream _stream = new();
+            private int _isClose;
+
+            public override WebSocketCloseStatus? CloseStatus { get; }
+            public override string CloseStatusDescription { get; }
+            public override WebSocketState State { get; }
+            public override string SubProtocol { get; }
+
+            public override async Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken)
+            {
+                if (_isClose == 1)
+                    throw new Exception("Closed");
+
+                await _stream.WriteAsync(buffer.Array, 0, buffer.Count, cancellationToken);
+            }
+
+            public async Task<string> CloseAndGetLogsAsync()
+            {
+                Close();
+                _stream.Seek(0, SeekOrigin.Begin);
+                using StreamReader reader = new(_stream);
+                return await reader.ReadToEndAsync();
+            }
+
+            public override void Dispose()
+            {
+                Close();
+                _stream.Dispose();
+            }
+
+            public override Task<WebSocketReceiveResult> ReceiveAsync(ArraySegment<byte> buffer, CancellationToken cancellationToken) =>_completionSource.Task;
+
+            private void Close()
+            {
+                if(Interlocked.CompareExchange(ref _isClose, 1, 0) == 0)
+                    _completionSource.SetResult(Result);
+            }
+
+            public override void Abort() { }
+            public override Task CloseAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken) => Task.CompletedTask;
+            public override Task CloseOutputAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken) => Task.CompletedTask;
+        }
+
+
         [Fact]
         public async Task DoNotCallUpdateLicenseLimitsCommandOnEveryLeaderChange()
         {
+            using var socket = new DummyWebSocket();
+
             var (servers, leader) = await CreateRaftCluster(3);
+            var _ = LoggingSource.Instance.Register(socket, new LoggingSource.WebSocketContext(), CancellationToken.None);
```

Comment:
How this is unregistered?

Reply:
We unregister when we try to send to the socket and get an error so the close inside `Dispose` and `CloseAndGetLogsAsync` handle that.
https://github.com/haludi/ravendb/blob/RavenDB-15494/src/Sparrow/Logging/LoggingSource.cs#L918
https://github.com/haludi/ravendb/blob/RavenDB-15494/src/Sparrow/Logging/LoggingSource.cs#L949